sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CallToAsyncMethodShouldNotBeBlocking.CSharp9.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Tests.Diagnostics
+{
+    public class Program
+    {
+        public Task Foo(Task<string> task)
+        {
+            Action<Task<string>, object> continuation;
+            return task.ContinueWith(state: null, continuationAction: (Task<string> _, object _) =>
+            {
+                Task<int> anotherTask = null; // Pretend to compute something
+                var b = anotherTask.Result; // Noncompliant, this task is not safe inside ContinueWith
```

Comment:
Theoretically we can call `.Result` also when using top-level statements. Shouldn't we add a test for that?